scraper.rb

```diff
@@ -1,290 +1,46 @@
-# although for people with only one term the page in question seems to fall over so
-# fall back to the current term and id for those people as it's presumably their first
-def get_unique_id(url, page_term, page_iddiputado, name)
-  cur_id = ScraperWiki::select('id FROM memberships WHERE iddiputado is ? AND term is ? and id <> 0', [page_iddiputado, page_term]) rescue nil
-  unless cur_id.nil? or cur_id.empty?
-    return cur_id[0][:id]
-  end
-  sleep(1)
-
-  visit url
-
-  term_map = {}
-  all('div.all_leg').each do |legislature|
-    within(legislature) do
-      term = nil
-      if legislature.has_css?('div.btn_ficha a')
-        link = find('div.btn_ficha a')
-        href = link['href']
-        # we can't do this as one operation as they don't always appear
-        # in the same order :(
-        term = href.to_s.match(/idLegislatura=(\d+)/).captures[0]
-        id = href.to_s.match(/idDiputado=(\d+)/).captures[0]
-        term_map[term.to_i] = id
-        save_membership_from_url(name, href)
-      end
-      if not term.nil? and legislature.has_css?('div.principal')
-        term_div = find('div.principal')
-        name, start_year, end_year = term_div.text.match(/(\w+\s*\w+)\s*\(\s*(\d+)\s*-\s*([^)]*)\)/).captures
-        if end_year.tidy == 'Actualidad'
-          end_year = ''
-        end
-        exists = ScraperWiki::select('id FROM terms WHERE id is ??', [id]) rescue nil
-        if exists.nil?
-          term = {
-            id: term,
-            name: name.tidy,
-            start_date: start_year.tidy,
-            end_date: end_year.tidy,
-            source: 'http://www.congreso.es/',
-          }
-          ScraperWiki.save_sqlite([:id], term, 'terms')
-        end
-      end
-    end
-  end
-
-  # the all terms page seems to be very unreliable so if we can't find what we expect
-  # then we should quite rather than trying to make up an incorrect ID
-  if term_map.empty?
-      return nil
-  end
-
-  min_term = term_map.keys.min
-
-  id = "#{min_term}_#{term_map[min_term]}"
-  for term in term_map.keys
-      ScraperWiki.sqliteexecute('update memberships set id = ? where id = 0 and term = ? and iddiputado = ?', [id, term, term_map[term]])
-  end
-  return id
-end
-
-def scrape_people(url)
-  visit url
-
-  all('div#RESULTADOS_DIPUTADOS div.listado_1 ul li a').each do |link|
-    save_membership_from_url(link.text, link['href'])
-  end
-
-  pagination = all('div.paginacion').first
-  next_page = nil
-  if pagination.has_xpath?(".//a[contains(.,'Página Siguiente')]")
-    within (pagination) do
-      next_page = find(:xpath, ".//a[contains(.,'Página Siguiente')]")
-    end
-  end
-
-  # the website is a bit fragile to lets not hammer it with requests
-  sleep(2)
-  unless next_page.nil?
-    scrape_people(next_page['href'])
+def person_data(data)
+  unwanted_keys = %i(party faction faction_id start_date end_date constituency term memberships_list)
```

Comment:
In general, if you're going to build a list solely for the purpose of looking things up in it, you should make it a Set instead. But the whole approach here seems slightly off to me, though I can't really tell what you're trying to do.